mods/give_initial_stuff/init.lua

```diff
@@ -1,12 +1,44 @@
-minetest.register_on_newplayer(function(player)
-	--print("on_newplayer")
-	if minetest.setting_getbool("give_initial_stuff") then
-		minetest.log("action", "Giving initial stuff to player "..player:get_player_name())
-		player:get_inventory():add_item('main', 'default:pick_steel')
-		player:get_inventory():add_item('main', 'default:torch 99')
-		player:get_inventory():add_item('main', 'default:axe_steel')
-		player:get_inventory():add_item('main', 'default:shovel_steel')
-		player:get_inventory():add_item('main', 'default:cobble 99')
+local stuff_string = minetest.setting_get("initial_stuff") or
+		"default:pick_steel,default:axe_steel,default:shovel_steel," ..
+		"default:torch 99,default:cobble 99"
+
+give_initial_stuff = {
+	items = {}
+}
+
+function give_initial_stuff.give(player)
+	minetest.log("action",
+			"Giving initial stuff to player " .. player:get_player_name())
+	local inv = player:get_inventory()
+	for _, stack in ipairs(give_initial_stuff.items) do
+		inv:add_item("main", stack)
+	end
```

Comment:
Will do.

My original thought was that it may be possible that this is not called when the inventory is empty. But I guess if a modder wanted to do that, they should make their own loop using give_initial_items:get_list()